File: eval/fixtures/js-eval/vulnerable.js
// eval/fixtures/js-eval/vulnerable.js
// BACO Eval Fixture: Code injection via eval() of user input (CWE-95)
// The vulnerability is on line 13 - eval() of req.query payload

const express = require('express');
const app = express();

// VULNERABLE: the query parameter is executed as JavaScript.
// An attacker can submit 'process.mainModule.require("child_process")...'
// as `expr` to run arbitrary code in the server process.
app.get('/calc', (req, res) => {
    const userExpr = req.query.expr;
    const result = eval(userExpr);
    res.json({ result });
});

File: eval/fixtures/py-cmdi/vulnerable.py
# eval/fixtures/py-cmdi/vulnerable.py
# BACO Eval Fixture: Command injection via os.system (CWE-78)
# The vulnerability is on line 16 - user input concatenated into shell command

import os


def ping_host(host: str) -> str:
    """
    Ping a host requested by the user.

    VULNERABLE: the host string is concatenated into a shell command.
    An attacker can supply "8.8.8.8; rm -rf /" to run arbitrary commands.
    """
    command = "ping -c 1 " + host
    exit_code = os.system(command)
    return f"ping exited with {exit_code}"
